util/compression_test.cc

```diff
@@ -1768,122 +1768,219 @@ TEST_F(DBAutoSkip, AutoSkipCompressionManager) {
     return true;
   }
 
  private:
-  int window_;
-  int num_keys_;
   const BlockBuilder& data_block_builder_;
-  IOCPUCostPredictor* predictor_;
 };
-class CostAwareTestFlushBlockPolicyFactory : public FlushBlockPolicyFactory {
+class AutoTuneFlushBlockPolicyFactory : public FlushBlockPolicyFactory {
  public:
-  explicit CostAwareTestFlushBlockPolicyFactory(const int window)
-      : window_(window) {}
+  explicit AutoTuneFlushBlockPolicyFactory() {}
 
   virtual const char* Name() const override {
-    return "CostAwareTestFlushBlockPolicyFactory";
+    return "AutoTuneFlushBlockPolicyFactory";
   }
 
   virtual FlushBlockPolicy* NewFlushBlockPolicy(
       const BlockBasedTableOptions& /*table_options*/,
       const BlockBuilder& data_block_builder) const override {
     (void)data_block_builder;
-    return new CostAwareTestFlushBlockPolicy(window_, data_block_builder);
+    return new AutoTuneFlushBlockPolicy(data_block_builder);
   }
-
- private:
-  int window_;
 };
-class DBCompressionCostPredictor : public DBTestBase {
+class DBAutoTuneCompressionTest : public DBTestBase {
  public:
-  Options options;
-  DBCompressionCostPredictor()
-      : DBTestBase("db_cpuio_skip", /*env_do_fsync=*/true),
-        options(CurrentOptions()) {
-    options.compression_manager = CreateCostAwareCompressionManager();
+  static constexpr double kCPUUpperBound = 0.7;
+  static constexpr double kCPULowerBound = 0.5;
+  static constexpr double kIOUpperBound = 0.7;
+  static constexpr double kIOLowerBound = 0.5;
+
+  Options options_;
+  Random rnd;
+  int next_key_;
+
+  DBAutoTuneCompressionTest()
+      : DBTestBase("db_autotune", /*env_do_fsync=*/true),
+        options_(CurrentOptions()),
+        rnd(231),
+        next_key_(0) {
     auto statistics = ROCKSDB_NAMESPACE::CreateDBStatistics();
-    options.statistics = statistics;
-    options.statistics->set_stats_level(StatsLevel::kExceptTimeForMutex);
+    options_.statistics = statistics;
+    options_.statistics->set_stats_level(StatsLevel::kExceptTimeForMutex);
     BlockBasedTableOptions bbto;
     bbto.enable_index_compression = false;
     bbto.flush_block_policy_factory.reset(
-        new CostAwareTestFlushBlockPolicyFactory(10));
-    options.table_factory.reset(NewBlockBasedTableFactory(bbto));
-    DestroyAndReopen(options);
-  }
-};
-TEST_F(DBCompressionCostPredictor, CostAwareCompressorManager) {
-  // making sure that the compression is supported
-  if (!ZSTD_Supported()) {
-    return;
+        new AutoTuneFlushBlockPolicyFactory());
+    options_.table_factory.reset(NewBlockBasedTableFactory(bbto));
+    options_.rate_limiter.reset(NewGenericRateLimiter(
+        1000000000, 1000 /* refill_period_us */, 10 /* fairness */,
+        RateLimiter::Mode::kWritesOnly));
+    std::shared_ptr<IOGoal> io_goal =
+        std::make_shared<IOGoal>(kIOUpperBound, kIOLowerBound);
+    std::shared_ptr<IOGoal> cpu_budget =
+        std::make_shared<IOGoal>(kCPUUpperBound, kCPULowerBound);
```

Comment:
If we can provide default value for budget, we don't even need to set these details in the unit test.